platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java

```diff
@@ -60,130 +62,246 @@
 
     private static final String OBJECT_NAME = CLASS_NAME + ":service=configuration-migration";
 
-    private final ConfigurationAdminMigration configurationAdminMigration;
+    private static final String PRODUCT_VERSION_FILENAME = "Version.txt";
+
+    private static final String EXPORT_EXTENSION = ".zip";
+
+    private static final String EXPORT_DATE_FORMAT = "-%tY%<tm%<tdT%<tH%<tM%<tS";
+
+    private static final String EXPORT_PREFIX = "exported-";
+
+    private static final String REBOOT_DELAY = "1"; // 1 minute
 
     private final MBeanServer mBeanServer;
 
-    private final List<ConfigurationMigratable> configurationMigratables;
+    private final List<Migratable> migratables;
+
+    private final SystemService system;
 
-    private final List<DataMigratable> dataMigratables;
+    private final String productVersion;
 
     /**
      * Constructor.
      *
-     * @param configurationAdminMigration object used to export {@link org.osgi.service.cm.Configuration}
-     *                                    objects from {@link org.osgi.service.cm.ConfigurationAdmin}
-     * @param mBeanServer                 object used to register this object as an MBean
-     * @param configurationMigratables    list of {@link ConfigurationMigratable} services. Needs
-     *                                    to be kept up-to-date by the client of this class.
-     * @param dataMigratables             list of {@link DataMigratable} services. Needs
-     *                                    to be kept up-to-date by the client of this class.
+     * @param mBeanServer object used to register this object as an MBean
+     * @param migratables list of {@link Migratable} services. Needs
+     *                    to be kept up-to-date by the client of this class.
+     * @param system      the system service
+     * @throws IOError if unable to load the distribution version information.
      */
-    public ConfigurationMigrationManager(
-            @NotNull ConfigurationAdminMigration configurationAdminMigration,
-            @NotNull MBeanServer mBeanServer,
-            @NotNull List<ConfigurationMigratable> configurationMigratables,
-            @NotNull List<DataMigratable> dataMigratables) {
-        notNull(configurationAdminMigration, "ConfigurationAdminMigration cannot be null");
-        notNull(mBeanServer, "MBeanServer cannot be null");
-        notNull(configurationMigratables,
-                "List of ConfigurationMigratable services cannot be null");
-        notNull(dataMigratables, "List of DataMigratable services cannot be null");
-
-        this.configurationAdminMigration = configurationAdminMigration;
+    public ConfigurationMigrationManager(MBeanServer mBeanServer, List<Migratable> migratables,
+            SystemService system) {
+        Validate.notNull(mBeanServer, "invalid null bean server");
+        Validate.notNull(migratables, "invalid null migratables");
```

Comment:
❓ Would it also be an error to have an empty list of `migratables`?

Reply:
It wasn't in the original code. It only means nothing would be exported ... not too useful. In any case, that should never happen as we have 4 system migratables registered (3 in this PR and 1 in an upcoming one)